sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_patch.py

```diff
@@ -0,0 +1,870 @@
+    """Implements a Location polling."""
+
+    def can_poll(self, pipeline_response: PipelineResponseType) -> bool:
+        """Answer if this polling method could be used.
+
+        :param pipeline_response: The PipelineResponse type
+        :type pipeline_response: PipelineResponseType
+        :return: Whether polling should be performed.
+        :rtype: bool
+        """
+        response = pipeline_response.http_response
+        can_poll = self._operation_location_header in response.headers
+        if can_poll:
+            return True
+
+        if not _is_empty(response):
+            body = _as_json(response)
+            status = body.get("status")
+            if status:
+                return True
+        return False
+
+    def _set_async_url_if_present(self, response: ResponseType) -> None:
+        location_header = response.headers.get(self._operation_location_header)
+        if location_header:
+            self._async_url = location_header
+        else:
+            self._async_url = response.request.url
+
+    def get_status(self, pipeline_response: PipelineResponseType) -> str:
+        """Process the latest status update retrieved from a 'location' header.
+
+        :param azure.core.pipeline.PipelineResponse pipeline_response: latest REST call response.
+        :return: The current operation status
+        :rtype: str
+        :raises: BadResponse if response has no body and not status 202.
+        """
+        response = pipeline_response.http_response
+        if not _is_empty(response):
+            body = _as_json(response)
+            status = body.get("status")
+            if status:
+                return self._map_nonstandard_statuses(status, body)
+            raise BadResponse("No status found in body")
+        raise BadResponse("The response from long running operation does not contain a body.")
+
+    def _map_nonstandard_statuses(self, status: str, body: Dict[str, Any]) -> str:
+        """Map non-standard statuses.
+
+        :param str status: lro process status.
+        :param str body: pipeline response body.
+        :return: The current operation status.
+        :rtype: str
+        """
+        if status == "ValidationFailed":
+            self.raise_error(body)
+        return status
+
+    def raise_error(self, body: Dict[str, Any]) -> None:
+        error = body["error"]
+        if body["error"].get("innerError", None):
+            error = body["error"]["innerError"]
+        http_response_error = HttpResponseError(message="({}): {}".format(error["code"], error["message"]))
+        http_response_error.error = ODataV4Format(error)  # set error.code
+        raise http_response_error
+
+
+class DocumentTranslationApiVersion(str, Enum, metaclass=CaseInsensitiveEnumMeta):
+    """Document Translation API versions supported by this package"""
+
+    #: This is the default version
+    V1_0 = "1.0"
+
+
+def validate_api_version(api_version: str) -> None:
```

Comment:
Are we still be using this method somewhere?

Reply:
It is not public and hence removed it